src/components/plotly/bar-line/index.js

```diff
@@ -8,71 +8,93 @@ import useTransformedData from '../shared/use-transformed-data'
 
 const BarLine = ({
   data,
+  x,
+  y,
+  y2,
   showTicks,
   showAxisTitles,
   axisTitles,
-  x,
-  y,
-  showCurrency,
+  showGrid,
   formatData,
   tickSuffix,
   tickPrefix,
   hoverInfo,
   hoverText,
   onAfterPlot,
+  chartOverlay,
+  lineWidth,
+  showLineMarkers,
+  sharedYAxis,
   ...props
 }) => {
-  const bar_data = {
+  const bar_ = useTransformedData({
     type: 'bar',
-    ...useTransformedData({
-      type: 'bar',
-      data,
-      x,
-      y: [y[0]],
-      orientation: 'v',
-      textPosition: 'none',
-      formatData,
-      tickSuffix,
-      tickPrefix,
-      hoverInfo: hoverInfo || 'x+text+name',
-      hoverText,
-      ...props,
-    })[0],
-  }
+    data,
+    x,
+    y,
+    orientation: 'v',
+    textPosition: 'none',
+    formatData,
+    tickSuffix: tickSuffix[0],
+    tickPrefix: tickPrefix[0],
+    hoverInfo: hoverInfo || 'x+text+name',
+    hoverText,
+    ...props,
+  })
 
-  const line_data = {
+  const line_ = useTransformedData({
     type: 'line',
-    yaxis: 'y2',
-    connectgaps: true,
-    ...useTransformedData({
-      type: 'line',
-      data,
-      x,
-      y: [y[1] ? y[1] : y[0]],
-      formatData,
-      tickSuffix,
-      tickPrefix,
-      hoverInfo: hoverInfo || 'x+text+name',
-      hoverText,
-      extra: {
-        mode: 'lines',
-        fill: 'tonexty',
+    data,
+    x,
+    y: y2,
+    formatData,
+    tickSuffix: tickSuffix[1],
+    tickPrefix: tickPrefix[1],
+    hoverInfo: hoverInfo || 'x+text+name',
+    hoverText,
+    extra: {
+      mode: showLineMarkers ? 'lines+markers' : 'lines',
+      line: {
+        width: lineWidth,
       },
-      ...props,
-    })[0],
+    },
+    ...props,
+  })
+
+  const getData = () => {
+    const dataArray = []
+
+    if (bar_.length) {
+      bar_.forEach(data => {
+        dataArray.push({
+          type: 'bar',
+          multi: 1,
+          ...data,
+        })
+      })
+    }
+    if (line_.length) {
+      line_.forEach(data => {
+        dataArray.push({
+          type: 'line',
+          yaxis: sharedYAxis ? '' : 'y2',
+          multi: 2,
+          ...data,
+        })
+      })
+    }
+
+    return dataArray
```

Comment:
Could be a bit more succinct with map, also this can be factored out of the component scope to be naturally memoized:

```jsx
const getDataArray = ({ bar, line }) => ([
  ...bar.map(data => ({ type: 'bar', multi: 1, ...data })),
  ...line.map(data => ({ type: 'line', ..., ...data })).
])

// inside component
  const bar = useTransformedData(...)
  const line = useTransformedData(...)

  return (
    <CustomPlot
      ...
      data={getDataArray({ bar, line })
    />
  )
```